### PR TITLE
[WIP] cephadm: add setup.py file for pip install

### PR DESCRIPTION
**Pull request by Daniel-Pivonka**
add setup.py file for pip install of cephadm

Signed-off-by: Daniel-Pivonka <dpivonka@redhat.com>

### CODE REVIEW

File: src/cephadm/setup.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"manages nodes in a cluster by establishing "
+	"an SSH connection and issuing explicit "
+	"management commands."
+	),
+     url="https://github.com/ceph/ceph/tree/master/src/cephadm",
+     license='LGPLv2+',
+     platforms='Linux',
+     packages=setuptools.find_packages(),
+     classifiers=[
+         "Programming Language :: Python :: 3",
```
Comment (tchaikov) [inline src/cephadm/setup.py]: cephadm is deliberately written so it's compatible to python2 and python3. could better off putting
```
         "Programming Language :: Python :: 2",
         "Programming Language :: Python :: 3",
```
here.

**Comment by sebastian-philipp** [comment]
Relates to #32526.

Problem here is, just adding a setup.py doesn't really solve anything:
* how do we distribute it? 
* How do we intend to users to install cephadm'? `sudo pip install 'git+https://github.com/ceph/ceph#egg=cephadm&subdirectory=src/cephadm'`? Or upload to pypi?
* How do we transfer cephadm in `root` mode to other hosts? The existing `cat cephadm | ssh` is dead-simple. In any case, this should IMO not install cephadm remotely. 
* how do we pass parameters and stdin to cephadm on remote machines?
* Are there going to be any weird interactions when also installing cephadm via RPM?
* CMake integration is missing
* How do you want to distribute cephadm within the RPM package? as before or do you want to make it a proper RPM-packaged python package (like python-common)?

**Comment by tchaikov** [comment]
yeah, after a second thought. `pip install` might not be a best way to redistribute cephadm..

**Comment by sebastian-philipp** [comment]
> yeah, after a second thought. `pip install` might not be a best way to redistribute cephadm..

Actually, I **do** think distributing via pip has many advantages that we should leverage, but we have to make it right and be sure to have answers for those questions.

**Comment by jmolmo** [comment]
My opinion/answers about the @sebastian-philipp questions:

> * how do we distribute it?

Upstream: why not PyPi?
Downstream: In the case of Red Hat possibly one rpm.

The thing i would love to do would be to use only PyPi.... (no rpms)

> * How do we intend to users to install cephadm'? `sudo pip install 'git+https://github.com/ceph/ceph#egg=cephadm&subdirectory=src/cephadm'`? Or upload to pypi?

I think that using PyPi has advantages, for example, we can control the version provided and when it is updated the pacjkage in the index. A link to one file in one repository can have unexpected consequences. (the same unexpected consequences as if you install something contained in a file in some place in some repo using a curl command)

> * How do we transfer cephadm in `root` mode to other hosts? The existing `cat cephadm | ssh` is dead-simple. In any case, this should IMO not install cephadm remotely.

In upstream, if we use PyPi we do not need to transfer nothing. 
In downstream, In the case of Red Hat we should have the rpm in the right repo.

> * how do we pass parameters and stdin to cephadm on remote machines?

Not understand very well. After cephadm is installed... what prevents you to execute commands remotely using ssh ? 

> * Are there going to be any weird interactions when also installing cephadm via RPM?

I think that this is a downstream issue: It will depend basically of what things we "put" into cephadm, and the way to build the cephadm rpm. Do we want to have an rpm distribution now? maybe this should be part of another PR.

> * CMake integration is missing

Sorry , new with cephadm , and the only thing i have seen is an unique python file...probably missing something... do we need this integration?

> * How do you want to distribute cephadm within the RPM package? as before or do you want to make it a proper RPM-packaged python package (like python-common)?

I think that we should build an RPM only for "cephadm" with the required dependencies. The RPM  used in Suse and RedHat is going to be the same?
I insist, i would love to install using only PyPi.

**Comment by sebastian-philipp** [comment]
Daniel, can you rebase this?

**Comment by Daniel-Pivonka** [comment]
closing as this is not the solution to the problem. it may be part of it but it will need to be revisited in the future

**Pull request closed without merging**